### PR TITLE
[VEGA-4996] (feat): add footerAlign prop in BaseModal

### PR DESCRIPTION
**Pull request by ndrzhn**
## Задача
Cсылка в Jira GPN на задачу: https://jira.konakov.tv/browse/VEGA-4996

## Problem statement

https://artcpt.atlassian.net/browse/VEGA-4996
нужна возможность менять выравнивание элементов в футере BaseModal

## Solution details

Добавлен prop footerAlign компоненту BaseModal, дефолтное значение - по правому краю.

## Type
- [x] Feature
- [ ] Bug Fix
- [ ] Refactoring
- [ ] Test
- [ ] Other

## Associated issues

_{Jira issues or PRs related to this one}_

## Quality control
- [x] PR: Based on a correct branch
- [x] PR: Head branch is rebased on the base branch
- [ ] PR: Assignee chosen and necessary labels are set
- [x] PR: Current form is filled out (where it makes sense)
- [x] PR: Diff checked for irrelevant changes
- [x] PR: Commit messages follow [the rules](../docs/commits-style.md)
- [x] JS: There's no errors/warnings in the browser dev console
- [x] Layout: Uses [variables](https://github.com/gpn-prototypes/ui-kit/tree/master/src/components/Theme) and [CSS classes](https://github.com/gpn-prototypes/ui-kit/blob/master/src/utils/whitepaper/whitepaper.css) from [GPN design system](https://github.com/gpn-prototypes/ui-kit)
- [x] Layout: Tested with various content amounts
- [ ] Docs: All changes to the component API are reflected and key points are described
- [ ] Storybook: Stories are written/updated for components
- [ ] Storybook: Stories follow [the rules](https://github.com/gpn-prototypes/vega-ui/blob/master/docs/storybook.md)
- [ ] Tests: New features and bugfixes are covered with tests
- [x] Tests: Existing tests passed successfully

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

### CODE REVIEW

**Review by Alexandr1111 — APPROVED** [review]

File: src/components/base-modal/BaseModal.tsx [hunk 1]
```diff
@@ -8,6 +8,10 @@ import './BaseModal.css';
 
 const cn = block('BaseModal');
 
+const footerPropAlign = ['right', 'center', 'left'] as const;
```
Comment (SergeyKatugin) [inline src/components/base-modal/BaseModal.tsx:11]: можно же просто type footerAlign = 'right' | 'left' | 'center';

File: src/components/base-modal/BaseModal.tsx [hunk 2]
```diff
@@ -8,6 +8,10 @@ import './BaseModal.css';
...
+type FooterPropAlign = typeof footerPropAlign[number];
+const footerPropAlignDefault: FooterPropAlign = footerPropAlign[0];
```
Comment (SergeyKatugin) [inline src/components/base-modal/BaseModal.tsx:13]: и тут `const defaultFooterAlign: footerAlign = 'right';`

Нейминг тоже поменять, чтоб без prop было

File: src/components/base-modal/BaseModal.tsx [hunk 3]
```diff
@@ -59,7 +65,11 @@ export const BaseModal: React.FC<Props> = ({
         )}
       </div>
 
-      <div className={cn('Footer')}>{footer}</div>
+      <div className={cn('Footer', {
+        right: footerAlign === footerPropAlign[0],
+        center: footerAlign === footerPropAlign[1],
+        left: footerAlign === footerPropAlign[2],
```
Comment (SergeyKatugin) [inline src/components/base-modal/BaseModal.tsx:71]: и тут уже будет `footerAlign === 'left'` и тп

Comment (SergeyKatugin) [reply src/components/base-modal/BaseModal.tsx:71]: и вынести в метод бы, чтоб в верстке логики поменьше было

File: src/components/base-modal/BaseModal.tsx (hunk 1 above)
Comment (ndrzhn) [reply src/components/base-modal/BaseModal.tsx:11]: поменяла

File: src/components/base-modal/BaseModal.tsx (hunk 3 above)
Comment (ndrzhn) [reply src/components/base-modal/BaseModal.tsx:71]: поменяла

**Review by SergeyKatugin — APPROVED** [review]

**Pull request merged**